### PR TITLE
Issue 12521

### PR DESCRIPTION
**Pull request by oarrietadotcms**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/util/ConfigUtils.java [hunk 1]
```diff
@@ -90,6 +90,18 @@ public static String getTimeMachinePath(){
 		return path;
 	}
 
+    public static String getStaticPublishPath(){
+
+        String path = Config.getStringProperty("STATIC_PUBLISHING_ROOT_PATH",
+                APILocator.getFileAssetAPI().getRealAssetsRootPath() + File.separator + "static_publishing");
+
+        File pathDir = new File(path);
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/util/ConfigUtils.java]: I think you might want to encapsulate this into the FileUtil.mkDirsIfNeeded (path)

File: dotCMS/src/main/java/com/dotcms/publisher/endpoint/ajax/PublishingEndpointAjaxAction.java [hunk 2]
```diff
@@ -170,19 +154,11 @@ public void editEndpoint(HttpServletRequest request, HttpServletResponse respons
...
-        		validatePublishingEndPointAWSS3(authKey);
-        	}
-            if (StaticPublisher.PROTOCOL_STATIC.equalsIgnoreCase(protocol)){
-                validatePublishingEndPointStatic();
-            }
-
-
-			PublishingEndPoint endpoint = new PublishingEndPoint();
-	        endpoint.setId(id);
+            final PublishingEndPointFactory factory = new PublishingEndPointFactory();
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/publisher/endpoint/ajax/PublishingEndpointAjaxAction.java]: we can have only one factory instance as a instance variable and re-use it

File: dotCMS/src/main/java/com/dotcms/publisher/endpoint/bean/factory/PublishingEndPointFactory.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+import com.dotcms.publisher.pusher.PushPublisher;
+import com.dotmarketing.util.UtilMethods;
+
+public class PublishingEndPointFactory {
+
+    public PublishingEndPoint getPublishingEndPoint(final String protocol) {
+        if (UtilMethods.isSet(protocol)) {
+            return null;
+        } else if (PushPublisher.PROTOCOL_HTTP.equalsIgnoreCase(protocol)
+                || PushPublisher.PROTOCOL_HTTP.equalsIgnoreCase(protocol)) {
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/publisher/endpoint/bean/factory/PublishingEndPointFactory.java]: was this for HTTP and HTTPS?

File: dotCMS/src/main/java/com/dotcms/publisher/endpoint/bean/PublishingEndPoint.java [hunk 4]
```diff
@@ -134,16 +136,15 @@ public StringBuilder toURL() {
...
-        } else {
-        	return PushPublisher.class;
-        }
-    }
+    public abstract Class getPublisher();
+
+    /**
+     * TODO
+     */
+    public abstract void validatePublishingEndPoint() throws DotDataException, LanguageException;
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/publisher/endpoint/bean/PublishingEndPoint.java]: after discussing with @jdotcms and @acurionedotcms, we think it is better to have one new "PublishingEndPointValidationException" that allows to add a list of un-i18n error messages (in a similar fashion to the validateContentlet), so this method will only that new (checked) exception

Comment (dsilvam) [reply dotCMS/src/main/java/com/dotcms/publisher/endpoint/bean/PublishingEndPoint.java]: users of the validate need to catch new the exception, get the list of error messages and get the i18n version of them using the user's info

**Review by dsilvam — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotmarketing/util/ConfigUtils.java (hunk 1 above)
Comment (oarrietadotcms) [reply dotCMS/src/main/java/com/dotmarketing/util/ConfigUtils.java]: @jdotcms I can't find that method.

**Review by dsilvam — APPROVED** [review]

**Pull request merged**